### DISCUSSION TITLE
Naming changes to avoid confusion

### DISCUSSION THREAD

**Discussion by 6DiegoDiego9** [GCuser99/SeleniumVBA#90] [General]
Two proposals of changes to reduce the likelihood of confusion:

1) rename "ImplicitWait" to "ImplicitTimeout", or "ImplicitWaitTimeout" (also used [here](https://www.selenium.dev/selenium/docs/api/java/org/openqa/selenium/WebDriver.Timeouts.html))
Proposal of description: "Poll the DOM for the given maximum time (in ms) when trying to find an element or elements if they are not immediately available"
2) merge ScrollToElement and ScrollIntoView: move ScrollToElement features ("x" and "y" offsets) to ScrollIntoView, and remove ScrollToElement  (or viceversa)

What do you think?

**Comment by GCuser99** [comment]
Diego, I'm good with those changes. I'm always hesitant to make user-facing structural changes but since we are doing it for IsPresent, might as well get all the potential pain over in one go... :-)

I prefer ImplicitWaitTimeout because it has the word "Wait" in the name and aligns with the cited standard. I like your description.

My slight preference is for keeping ScrollIntoView and eleminating ScrollToElement. Agree that they are partially redundant.

**Comment by 6DiegoDiego9** [comment]
My experience with ImplicitWait with a couple of colleagues that are new to Selenium and VBA, is that their first thought is that it adds a wait for the (entire) given time, then (tries to) finds the element(s).
I also find the description in my link confusing in this direction.
I'd like to avoid this confusion.

ImplicitWaitTimeout would be ok for me. An alternative could be ImplicitMaxWait.

**Comment by GCuser99** [comment]
Hi @6DiegoDiego9, I can't get ScrollIntoView to work when the enSpeed argument is set to jump_smooth.  It fails to scroll on Chrome and Edge. Works fine for Firefox. IE mode scrolls but not smoothly and takes a long time. Were you able to get it to work? A quick Google search yields many bug reports for Chrome/Edge.

It's not that important - just wanted to check if you found a way...

```vba
Sub test_scrollIntoView()
    Dim driver As New WebDriver
    Dim webElem As WebElement
 
    driver.StartEdge
    driver.OpenBrowser

    driver.NavigateTo "http://demo.guru99.com/test/guru99home/"
    driver.ActiveWindow.Maximize
    driver.Wait 1000
     
    driver.FindElement(By.LinkText, "Linux").ScrollIntoView enSpeed:=jump_smooth, enAlign_vert:=align_start
    'driver.FindElement(By.LinkText, "Linux").ScrollIntoView enSpeed:=jump_auto, enAlign_vert:=align_start
    
    driver.Wait 3000
        
    driver.CloseBrowser
    driver.Shutdown
End Sub
```

Mike

**Comment by 6DiegoDiego9** [comment]
It seems working to me:

[SeleniumVBA.webm](https://github.com/GCuser99/SeleniumVBA/assets/109476941/6b69d000-3cbb-4dda-988a-5a3661e489e8)

**Comment by GCuser99** [comment]
Thx Diego. I'll just have to treat it as an unsolved mystery why it isn't working for me... I'll publish the next version with changes you requested within a few days...
